benchmarks/synchrobench/src/main/java/com/oath/oak/synchrobench/contention/benchmark/Test.java

```diff
@@ -13,441 +13,478 @@
-		Test test = new Test(args);
-		test.printParams();
-
-		// warming up the JVM
-		if (Parameters.warmUp != 0) {
-			try {
-				test.initThreads();
-			} catch (Exception e) {
-				System.err.println("Cannot launch operations.");
-				e.printStackTrace();
-			}
-			test.execute(Parameters.warmUp * 1000, true);
-			// give time to the JIT
-			Thread.sleep(1000);
-			if (Parameters.detailedStats)
-				test.recordPreliminaryStats();
-			test.clear();
-			test.resetStats();
+                oakBench = (CompositionalOakMap<MyBuffer, MyBuffer>) c.newInstance();
+                benchType = Type.OAKMAP;
+            }
+
+        } catch (Exception e) {
+            System.err.println("Cannot find benchmark class: " + benchName);
+            System.exit(-1);
+        }
+    }
+
+
+    /**
+     * Creates as many threads as requested
+     *
+     * @throws InterruptedException if unable to launch them
+     */
+    private void initThreads() throws InterruptedException {
+        switch (benchType) {
+            case OAKMAP:
+                threadLoopsOak = new ThreadLoopOak[Parameters.numThreads];
+                threads = new Thread[Parameters.numThreads];
+                for (short threadNum = 0; threadNum < Parameters.numThreads; threadNum++) {
+                    threadLoopsOak[threadNum] = new ThreadLoopOak(threadNum, oakBench, methods);
+                    threads[threadNum] = new Thread(threadLoopsOak[threadNum]);
+                }
+                break;
+        }
+    }
+
+    /**
+     * Constructor sets up the benchmark by reading parameters and creating
+     * threads
+     *
+     * @param args the arguments of the command-line
+     */
+    public Test(String[] args) {
+        printHeader();
+        try {
+            parseCommandLineParameters(args);
+        } catch (Exception e) {
+            System.err.println("Cannot parse parameters.");
+            e.printStackTrace();
+        }
+        instanciateAbstraction(Parameters.benchClassName);
+        this.throughput = new double[Parameters.iterations];
+        this.totalSize = new int[Parameters.iterations];
+    }
+
+    private void printHeapStats(String message) {
+        System.gc();
+        long heapSize = Runtime.getRuntime().totalMemory(); // Get current size of heap in bytes
+        long heapFreeSize = Runtime.getRuntime().freeMemory();
+
+        System.out.println("\n" + message);
+        System.out.println((float) (heapSize - heapFreeSize) / (1024 * 1024));
+        System.out.println((float) (((OakMap) oakBench).ma.allocated()) / (1024 * 1024));
+    }
+
+    /**
+     * Execute the main thread that starts and terminates the benchmark threads
+     *
+     * @throws InterruptedException
+     */
+    private void execute(int milliseconds, boolean maint)
+            throws InterruptedException {
+        printHeapStats("Before");
+        long startTime = System.currentTimeMillis();
+        long count = fill(Parameters.range, Parameters.size);
```

Comment:
Why do you count all operations (and not successes of putIfAbsent)? Isn't it just size? Why is it interesting to print it?

Reply:
We count the operations to measure the performance of filling up the index.
Since not every `putIfAbsent` succeeds, but it is still and operation that was executed, we count it.
Otherwise, it would not reflect the actual throughput, but rather the randomness of each run.

Reply:
But fill() method does exactly Parameters.size operations in its for-loop. You have the number, why to count?

Reply:
It may do more `putIfAbsent` if the key it draws already exists